app/repositories/SessionRepository.scala

```diff
@@ -16,67 +16,45 @@
 
 package repositories
 
-import java.time.LocalDateTime
-
-import javax.inject.Inject
 import models.UserAnswers
+import org.mongodb.scala.model.Indexes.ascending
+import org.mongodb.scala.model.{Filters, IndexModel, IndexOptions, ReplaceOptions}
 import play.api.Configuration
-import play.api.libs.json._
-import play.modules.reactivemongo.ReactiveMongoApi
-import reactivemongo.api.indexes.IndexType
-import reactivemongo.play.json.collection.Helpers.idWrites
-import reactivemongo.play.json.collection.JSONCollection
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
 
+import java.time.LocalDateTime
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class DefaultSessionRepository @Inject()(
-                                          val mongo: ReactiveMongoApi,
-                                          val config: Configuration
-                                        )(implicit val ec: ExecutionContext)
-  extends SessionRepository
-    with IndexManager {
-
-  override val collectionName: String = "user-answers"
-
-  private val cacheTtl = config.get[Int]("mongodb.timeToLiveInSeconds")
-
-  private def collection: Future[JSONCollection] = for {
-    _ <- ensureIndexes
-    col <- mongo.database.map(_.collection[JSONCollection](collectionName))
-  } yield col
-
-  private val lastUpdatedIndex = MongoIndex(
-    key = Seq("lastUpdated" -> IndexType.Ascending),
-    name = "user-answers-last-updated-index",
-    expireAfterSeconds = Some(cacheTtl)
+class DefaultSessionRepository @Inject()(val mongo: MongoComponent, val config: Configuration)(implicit val ec: ExecutionContext
+) extends PlayMongoRepository[UserAnswers](
+    collectionName = "user-answers",
+    mongoComponent = mongo,
+    domainFormat = UserAnswers.userAnswersFormat,
+    indexes = Seq(
+      IndexModel(
+        ascending("lastUpdated"),
+        indexOptions = IndexOptions()
+          .name("user-answers-last-updated-index")
+          .expireAfter(config.get[Int]("mongodb.timeToLiveInSeconds"), TimeUnit.SECONDS)
+      )
```

Comment:
where is drop indexes flag here? you had it in Indexmanager so we need to add that replaceIndex param here and use that flag from app.conf